docs: new readme (#70)
stoplightio/http-spec: README.md

# @stoplight/http-spec

> ## What is this

Reviewer: ```suggestion
## What is it?
```

> Http Spec is yet another, hopefully more pragmatic, attempt to standardise OpenAPI 2, OpenAPI 3, Postman Collections and other HTTP-related specification under a single AST to rule them all (at least in Stoplight)

Reviewer: Should "HTTP" be capitalized in all cases?

```suggestion
HTTP Spec is yet another, hopefully more pragmatic, attempt to standardise OpenAPI v2, OpenAPI v3, Postman Collections and other HTTP-related specification under a single AST to rule them all (at least in Stoplight).
```

> There have been a lot of other attempts to have this universal thing such as API Elements, API Flow and they all failed for a number of reason, some of them have been discussed [here](https://www.youtube.com/watch?v=jn_1lJr-DLY)

Reviewer: ```suggestion
There have been a lot of other attempts to have a universal specification such as API Elements, API Flow and they all failed for a number of reasons. Some of those have been discussed [here](https://www.youtube.com/watch?v=jn_1lJr-DLY).
```

> ## So why doing this?

Reviewer: ```suggestion
## Why build it?
```

> Because we still need a way to interact with these documents in a standardised way, and relying on some dying intermediate format or something that neither the creators are using anymore does not make that much sense. It's better to accepts the sad state of the industry and work on the a mininal superset format that can encompass the majority of the use cases.

Reviewer: ```suggestion
Stoplight needs a way to interact with these documents in a standardized way, and relying on some dying intermediate format does not make that much sense. It's better to accept the sad state of the industry and work on a minimal superset format that can encompass the majority of the use cases.
```

> ## How's Http Spec formed?

Reviewer: ```suggestion
## How's HTTP Spec formed?
```

> This repository contains *exclusively* the converters, which are the functions that take an OpenAPI2/3 document or a Postman Collection and turns it out into a tree whose structure is defined in another package, [`@stoplight/types`](https://github.com/stoplightio/types), specifically in the [http-spec file](https://github.com/stoplightio/types/blob/master/src/http-spec.ts)

Reviewer: ```suggestion
This repository contains *exclusively* converters functions that take OpenAPI v2, OpenAPI v3, or Postman Collection documents and transforms them into the [http-spec interface](https://github.com/stoplightio/types/blob/master/src/http-spec.ts).
```

You can explore the whole structure by looking at the [IHttpService](https://github.com/stoplightio/types/blob/master/src/http-spec.ts#L10) definition and checking out its descendands. You'll probably notice that it resembles a lot the current OpenAPI 3.x specification, and that's on purpose. OpenAPI 3.0 has first support and we gracefully upgrade/downgrade the other specification formats to it.

> ## How do I write a converter

Reviewer: ```suggestion
## How do I write a converter?
```

> With such foreword, the steps should be pretty clear:

Reviewer 2: Saying things like "simple" "clear" "easy" can make people feel a bit daft if it is not clear to them. 

```suggestion
If you would like to add support for another API description format, like RAML, follow these steps:
```

> 1. Create a new directory in the `src/` directory

Reviewer 2: This contributing guide seems to make it harder for general contributors who might be interested in making some changes, running the test suite, and committing back. Is your guide specifically just for people want to add a new set of transformers, or something else?

Author: I personally take the absence of such thing as a "this repository has nothing different than a regular NodeJs repo" — and so all the commands you'd use everywhere will just work. I can put it back, but I do not think it will add that much value.

2-3 times I ended up reading this whole thing just to realise that it's just how it should work by default.

Reviewer 2: I hear you, but npm vs yarn is not always clear and can lead to problems. yarn test.prod is not default and can lead to problems. 

Beyond that yes it's a bit of a pain in the backside to include "this is how you do a PR" but if we can link to blog posts instead then great. If we dont need yarn commit then we dont need to talk about that, but if we do need it then we do need to talk about it.

We need to treat this like an OSS project because some day it might be. It's public after all, so technically it already is.

Author: > yarn test.prod

And indeed should not be used, that's a command used by the CI.

>  If we dont need yarn commit

We do not, we have husky that does the same job. I'll actually remove all this stuff from package.json as well.

Reviewer 2: Great! The easier this all is the easier the Contributing instructions can be. So long as we're telling people the truth here its all good.

2. Create a function that from your input returns an `IHttpService`

> 3. Let TypeScript's error guide you while filling the missing fields (suchs as Security Scheems, Servers)

Reviewer: ```suggestion
3. Let the TypeScript errors guide you while filling out the missing fields (such as Security Schemes, Servers)
```
